pkg/cas/hardlinking_file_fetcher.go

```diff
@@ -121,7 +125,7 @@ func (ff *hardlinkingFileFetcher) GetFile(ctx context.Context, blobDigest digest
 	} else if wasMissing {
 		// Even though the file is part of our bookkeeping, we
 		// observed it didn't exist. Repair this inconsistency.
-		if err := directory.Link(name, ff.cacheDirectory, path.MustNewComponent(key)); err != nil && !os.IsExist(err) {
+		if err := ff.linkHelper.Link(directory, name, ff.cacheDirectory, path.MustNewComponent(key)); err != nil && !os.IsExist(err) {
```

Comment:
Does `Clonefile()` preserve the file modification time, or does the cloned instance get its modification time updated? In case it's the latter, do we want to add an additional call to reset it back to Jan 1st, 2000? Some build actions (mainly ones using rules_foreign_cc) get annoyed if we don't ensure all input files have the same modification time.

Reply:
An unscientific check using `cp -c` shows the mod time preserved (whereas normal cp does not).

Although I have decided that the correct answer when using rules_foreign_cc is "burn it with fire"